controllers/profiles/reconciler_dev.go

```diff
@@ -180,8 +220,10 @@ func (f *followDeployDevWorkflowReconciliationState) Do(ctx context.Context, wor
 	if kubeutil.IsDeploymentAvailable(deployment) {
 		workflow.Status.Manager().MarkTrue(api.RunningConditionType)
 		f.logger.Info("Workflow is in Running Condition")
+		f.enrichers.workflowStatus.Enrich(ctx, workflow)
```

Comment:
Is this a post-processing, on-exit state of some sort? If so, we can treat it as this or just do what we need within the state, hard to identify. I think it's a better approach so we don't struggle to debug a problem caused by something happening after we exit the state. If we do an on-exit or post-processing, we should make it clear in the state API "Do, CanReconcile, OnExit".. etc.

Reply:
I refactore the code based on your feedback: 
- added a PostReconcile function
- the function will be executed after the Do and won't affect the reconcile result so that we can put there operations that are not mandatory for the reconciliation like the status enrichment
- I put a default PostReconcile func into the stateSupport struct so that we can use the null-pattern: we don't need to implement it for all the states and at the same time we don't need to put conditions inside the code 

wdyt?